tools/scylla-sstable.cc

```diff
@@ -1726,7 +1726,31 @@ void decompress_operation(schema_ptr schema, reader_permit permit, const std::ve
     }
 }
 
+} // anonymous namespace
+
+namespace tools {
+
+/// Parse a stream of JSON encoded mutation fragments, using the format produced
+/// by scylla-sstable write.
+/// The parsing is streamed, at most a single mutation fragment is kept in memory.
+/// Not all data types are supported: collections, tuples, UDTs and counters are not supported.
+/// See docs/operating-scylla/admin-tools/scylla-sstable.rst for the format documentation.
 class json_mutation_stream_parser {
+    class impl;
+    std::unique_ptr<impl> _impl;
+
+public:
+    explicit json_mutation_stream_parser(schema_ptr schema, reader_permit permit, input_stream<char> istream, logger& logger);
+    json_mutation_stream_parser(json_mutation_stream_parser&&);
```

Comment:
noexcept (typically important for move constructors). Can also use = default.

Reply:
Actually, out-of-line definition is needed, as the move-constructor evaluates `~unique_ptr()`.
But I will add the noexcept.

Reply:
Why would the move constructor destroy anything?

Reply:
Yes, it is enough to evaluate the destructor, even if on a branch which is not taken.